crates/rpc/src/handlers/validator.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::Arc;
+
+use ream_bls::pubkey::pubkey_from_str;
+use ream_storage::db::ReamDB;
+use serde_json::json;
+use warp::{
+    http::status::StatusCode,
+    reject::Rejection,
+    reply::{self, Reply, with_status},
+};
+
+use super::{BeaconResponse, state::get_state_from_id};
+
+pub async fn get_validator_from_state(
+    state_id: String,
+    validator_id: String,
+    db: Arc<ReamDB>,
+) -> Result<impl Reply, Rejection> {
+    let validator_key = match pubkey_from_str(&validator_id) {
+        Ok(key) => key,
+        Err(_) => {
+            let error_body = reply::json(&json!({
+                "code": 400,
+                "message": format!("Invalid validator ID: {}", validator_id),
+            }));
```

Comment:
you should make an enum for errors don't create raw errors like this, you will be using these error codes again a lot in followup pr's